Sources/EventProducer/TidalEventSender.swift

```diff
@@ -29,29 +29,36 @@ public final class TidalEventSender: EventSender {
 	private var scheduler: EventScheduler
 	private let eventSubmitter: EventSubmitter
 	private var monitoring: Monitoring
+	private let eventQueue: EventQueue
+	private let monitoringQueue: MonitoringQueue
 	private var monitoringScheduler: MonitoringScheduler
 	private let fileManager: FileManagerHelper
 	
 	// MARK: - Init
 	public init() {
-		self.eventSubmitter = EventSubmitter.shared
-		self.monitoring = Monitoring.shared
-		self.scheduler = EventScheduler(consumerUri: config?.consumerUri)
-		self.monitoringScheduler = MonitoringScheduler(consumerUri: config?.consumerUri)
+		self.eventQueue = EventQueue()
+		self.monitoringQueue = MonitoringQueue()
+		self.monitoring = Monitoring(monitoringQueue: monitoringQueue)
+		self.eventSubmitter = EventSubmitter(eventsQueue: eventQueue, monitoring: monitoring)
+		
 		self.fileManager = FileManagerHelper.shared
+		
+		self.scheduler = .init(config: config, eventQueue: eventQueue, monitoring: monitoring)
+		self.monitoringScheduler = .init(config: config, eventQueue: eventQueue, monitoring: monitoring)
```

Comment:
At this point, to my understanding the config is `nil` and the `scheduler` won't do anything either way. Probably worth making scheduler optional (or `!`)? So that people don't forget to set config? 🤔

Reply:
we can do something similar to `TidalAuth` implementation - throw an error if a config was not set (and schedulers were not initialized). 

But I wouldn't do it here, as it is not really related. Let's maybe make it in a separate PR?

Reply:
Sure, if you prefer this way.